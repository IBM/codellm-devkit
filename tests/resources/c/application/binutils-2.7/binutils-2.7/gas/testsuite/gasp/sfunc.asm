
	.MACRO	RESERVE_STR P1=0 P2
	.SDATA	.SUBSTR("ABCDEFG",\P1,\P2)
	.ENDM

	RESERVE_STR	2,2
	RESERVE_STR	,3


	.MACRO FIND_STR P1
	.DATA.W	.INSTR("ABCDEFG","\P1", 0)
	.ENDM

	FIND_STR	CDE
	FIND_STR	H

	.MACRO	RESERVE_LENGTH P1
	.ALIGN	4
	.SRES	.LEN("\P1")
	.ENDM

	RESERVE_LENGTH	ABCDEF
	RESERVE_LENGTH	ABC

	.END

