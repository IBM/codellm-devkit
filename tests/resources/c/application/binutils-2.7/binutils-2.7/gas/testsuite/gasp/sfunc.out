!

!	.MACRO	RESERVE_STR P1=0 P2
!	.SDATA	.SUBSTR("ABCDEFG",\P1,\P2)
!	.ENDM
!

!	RESERVE_STR	2,2
!	.SDATA	.SUBSTR("ABCDEFG",2,2)
	.byte	67,68
!	RESERVE_STR	,3
!	.SDATA	.SUBSTR("ABCDEFG",0,3)
	.byte	65,66,67
!

!

!	.MACRO FIND_STR P1
!	.DATA.W	.INSTR("ABCDEFG","\P1", 0)
!	.ENDM
!

!	FIND_STR	CDE
!	.DATA.W	.INSTR("ABCDEFG","CDE", 0)
	.short	2
!	FIND_STR	H
!	.DATA.W	.INSTR("ABCDEFG","H", 0)
	.short	-1
!

!	.MACRO	RESERVE_LENGTH P1
!	.ALIGN	4
!	.SRES	.LEN("\P1")
!	.ENDM
!

!	RESERVE_LENGTH	ABCDEF
!	.ALIGN	4
	.align	4
!	.SRES	.LEN("ABCDEF")
	.space	24
!	RESERVE_LENGTH	ABC
!	.ALIGN	4
	.align	4
!	.SRES	.LEN("ABC")
	.space	12
!

!	.END
